lib/std/simd.zig

```diff
@@ -0,0 +1,369 @@
+// Note: Some functions here are known to malfunction on MIPS.
+// Functions whose tests fail on MIPS will trigger compile errors when compiled for that platform.
+
+const std = @import("std");
+const builtin = @import("builtin");
+
+pub const Vector = std.meta.Vector;
+
+/// Suggests a vector size for a given type and CPU, or null if it doesn't support SIMD natively.
+/// Not yet implemented for every CPU architecture.
+pub fn suggestVectorSizeForCpu(comptime T: type, cpu: std.Target.Cpu) ?usize {
+    switch (cpu.arch) {
+        .x86_64 => {
+            if (T == bool and std.Target.x86.featureSetHas(.prefer_mask_registers)) return 64;
+
+            const vector_bit_size = if (std.Target.x86.featureSetHas(.avx512f)) 512 else if (std.Target.x86.featureSetHas(.prefer_256_bit)) 256 else if (std.Target.x86.featureSetHas(.prefer_128_bit)) 128 else return null;
```

Comment:
`else return unreachable`
Also, please split this in more lines.

Reply:
Can I split this line up without zig fmt reverting it?

Reply:
using brackets should work. Not sure, if comma like for function arguments also works.